### PR TITLE
Correctly parse all URL schemes in UrlHelper::isLookLikeUrl().

### LINKED ISSUE #8722: Tracking outgoing links other than http or https is broken
I'm using the normal tracking code from Piwiks admin page with link tracking enabled. Outgoing links are normally recorded by Piwik. All fine.

But I also have links other than HTTP or HTTPS on my website, for example, rtp: // ... These links will be logged, but misrepresented on the Piwik page. They all appear as a link to the Piwik Dashboard. In the visitor profile these links as shown as "Page URL not defined". Screenshot attached.

screenshot-piwik tausys de 2015-09-07 11-11-49

### PR DESCRIPTION
**Pull request by diosmosis**
As title. Changed the regex to match all allowed schemes (as per the RFC: https://tools.ietf.org/html/rfc3986#section-3.1).

Fixes #8722

### CODE REVIEW

**Comment by sgiehl** [comment]
There are two integration tests failing now. Maybe we should check where the `isLookLikeUrl` method is used. At some points maybe it would be more correct to really check for an URL

**Comment by diosmosis** [comment]
Fixed the failing tests. Will fix other failures if they show up.

> At some points maybe it would be more correct to really check for an URL

We should probably replace this w/ a utility library in 3.0. For LTS, big changes like this would be counterproductive.

**Comment by diosmosis** [comment]
Failing UI test is a random failure

File: tests/UI/specs/Installation_spec.js [hunk 1]
```diff
@@ -112,8 +112,10 @@ describe("Installation", function () {
     it("should display the javascript tracking page when correct information is entered in the setup website page and next is clicked", function (done) {
         expect.screenshot("js_tracking").to.be.capture(function (page) {
             page.sendKeys('input[name=siteName]', 'Serenity');
-            page.sendKeys('input[name=url]', 'serenity.com');
             page.evaluate(function () {
+                // cannot use sendKeys since quickform does not use placeholder attribute
```
Comment (mattab) [inline tests/UI/specs/Installation_spec.js]: @diosmosis curious why it's still above `page.sendKeys('input[name=siteName]', 'Serenity');` if we cannot use sendKeys?

**Comment by mattab** [comment]
LGTM and works!

File: tests/UI/specs/Installation_spec.js (hunk 1 above)
Comment (diosmosis) [inline tests/UI/specs/Installation_spec.js]: Forgot to delete it :)

**Pull request merged**